nflow-engine/src/main/java/com/nitorcreations/nflow/engine/internal/dao/DaoUtil.java

```diff
@@ -35,4 +38,21 @@ public static Integer getInt(ResultSet rs, String columnLabel) throws SQLExcepti
     int value = rs.getInt(columnLabel);
     return rs.wasNull() ? null : value;
   }
+
+  public static final class ColumnNamesExtractor implements ResultSetExtractor<List<String>> {
+    static final ColumnNamesExtractor columnNamesExtractor = new ColumnNamesExtractor();
+
+    private ColumnNamesExtractor() {
+    }
+
+    @Override
+    public List<String> extractData(ResultSet rs) throws SQLException, DataAccessException {
+      List<String> columnNames = new LinkedList<>();
```

Comment:
arraylist instead? we're not going to remove column names from the middle of the list?